static/app/views/performance/newTraceDetails/traceApi/utils.tsx

```diff
@@ -8,3 +16,81 @@ export function isEmptyTrace(trace: TraceTree.Trace): boolean {
 
   return trace.length === 0;
 }
+
+const CANDIDATE_TRACE_TITLE_OPS = ['pageload', 'navigation', 'ui.load'];
+
+export type RepresentativeTraceEvent = {
+  event: TraceTree.TraceEvent | OurLogsResponseItem | null;
+  type: 'trace' | 'log';
+};
+
+export const getRepresentativeTraceEvent = (
+  tree: TraceTree,
+  logs: OurLogsResponseItem[] | undefined
+): RepresentativeTraceEvent => {
+  const hasLogs = logs && logs.length > 0;
+  if (tree.type === 'empty' && hasLogs) {
+    return {
+      event: logs[0]!,
+      type: 'log',
+    };
+  }
+
+  const traceNode = tree.root.children[0];
+
+  if (!traceNode) {
+    return {
+      event: null,
+      type: 'trace',
+    };
+  }
+
+  if (!isTraceNode(traceNode) && !isEAPTraceNode(traceNode)) {
+    throw new TypeError('Not trace node');
+  }
+
+  let firstRootEvent: TraceTree.TraceEvent | null = null;
+  let candidateEvent: TraceTree.TraceEvent | null = null;
+  let firstEvent: TraceTree.TraceEvent | null = null;
+
+  const events = isTraceNode(traceNode)
+    ? [...traceNode.value.transactions, ...traceNode.value.orphan_errors]
+    : traceNode.value;
+  for (const event of events) {
+    // If we find a root transaction, we can stop looking and use it for the title.
+    if (!firstRootEvent && isRootEvent(event)) {
+      firstRootEvent = event;
+      break;
+    } else if (
+      // If we haven't found a root transaction, but we found a candidate transaction
+      // with an op that we care about, we can use it for the title. We keep looking for
+      // a root.
+      !candidateEvent &&
+      CANDIDATE_TRACE_TITLE_OPS.includes(
+        'transaction.op' in event
+          ? event['transaction.op']
+          : 'op' in event
+            ? event.op
+            : ''
+      )
+    ) {
+      candidateEvent = event;
+      continue;
```

Comment:
should we just use `break` instead? if we're returning `event: firstRootEvent ?? candidateEvent ?? firstEvent,` then we will never use `firstEvent` anyways if we have a `candidateEvent`.

Reply:
@edwardgou-sentry but we might find a rootEvent after we find a candidate